packages/modules/backup_clouds/samba/backup_cloud.py

```diff
@@ -8,16 +12,44 @@
 
 log = logging.getLogger(__name__)
 
+def isPortOpen(host, port):
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(3)
+        try:
+                s.connect((host, int(port)))
+                s.shutdown(socket.SHUT_RDWR)
+                return True
+        except:
+                return False
+        finally:
+                s.close()
 
 def upload_backup(config: SambaBackupCloudConfiguration, backup_filename: str, backup_file: bytes) -> None:
-    conn = SMBConnection(config.smb_user, config.smb_password, 'client', 'server', use_ntlm_v2=True)
-    conn.connect(config.smb_path, 445)
-
-    with conn.openFile(backup_filename, 'w') as file:
-        file.write(backup_file)
-
-    conn.close()
-
+    conn = SMBConnection(config.smb_user, config.smb_password, os.uname()[1], config.smb_server, use_ntlm_v2=True)
+    foundedChars = re.search(r'[\\\:\*\?\"\<\>\|]+', config.smb_path)
```

Comment:
Bitte die Python Namenskonventionen beachten. Wörter in Variablen werden mit Unterstrichen verbunden, nicht CamelCase. Als Beispiel für diese Zeile:
```suggestion
    found_invalid_chars = re.search(r'[\\\:\*\?\"\<\>\|]+', config.smb_path)
```
Weiterhin ist der Name "founded chars" nicht richtig. Es müsste "found chars" oder besser "found invalid chars" heißen.
Weitere Änderungsvorschläge:
`isHostReachable` -> `host_is_reachable`
`sendFile` -> `send_file`

Reply:
verstanden und umgesetzt